### PR TITLE
(refactor): Propagating the kubeconfig of the newly created GKE cluster into experiment stages

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: udit uditgaurav@gmail.com

- Propagating the kubeconfig of the newly created GCP into experiment stages
- Use GKE cluster as the target cluster environment

### CODE REVIEW

File: utils/chaosengine.j2 [hunk 1]
```diff
@@ -5,11 +5,19 @@ metadata:
   name: {{ name }}
   namespace: litmus
 spec:
+<<<<<<< HEAD
+=======
+<<<<<<< HEAD
+=======
+>>>>>>> master
```
Comment (chandankumar4) [inline utils/chaosengine.j2]: This is not required, While rebasing the code it might came

File: build/gitlab/stages/create-gke-cluster.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            executable: /bin/bash
+            warn: False
+
+        - name: Select a region to create the cluster
+          set_fact: 
+            set_zone: "{{ item }}"
+          with_random_choice: "{{ zone }}"
+
+        - name: create GKE cluster
+          shell: gcloud container clusters create {{ cluster_name }}  --enable-autoscaling --max-nodes=5 --min-nodes=3 --machine-type=n1-standard-4 --disk-size=80 --image-type=ubuntu --zone {{ set_zone }}  --project={{ project }}  --scopes=https://www.googleapis.com/auth/cloud-platform
```
Comment (ksatchit) [inline build/gitlab/stages/create-gke-cluster.yml]:
```yaml
shell: > 
  - <>
```

File: experiments/generic/container-kill.yml [hunk 3]
```diff
@@ -12,30 +12,37 @@
...
       shell: kubectl get po {{ name }}-runner -n {{ a_ns }} -o custom-columns=:.status.phase --no-headers
       register: result
+      args:
+        executable: /bin/bash
       until: "'Succeeded' in result.stdout"
       delay: 10
       retries: 50
 
     - name: Getting chaos-result
       shell: kubectl get chaosresult -n litmus
```
Comment (ksatchit) [inline experiments/generic/container-kill.yml]: have apps in different ns & chaos in the respective ns

File: utils/chaosengine.j2 [hunk 4]
```diff
@@ -9,7 +9,6 @@ spec:
     runner:
       image: "litmuschaos/chaos-executor:ci"
       type: "go"   
-  jobCleanUpPolicy: retain
   monitoring: false
```
Comment (chandankumar4) [inline utils/chaosengine.j2]: is `jobCleanUpPolicy: retain` not required?

Comment (chandankumar4) [reply utils/chaosengine.j2]: You can set it to delete also

Comment (ksatchit) [reply utils/chaosengine.j2]: yes!

**Pull request merged**